src/Symfony/Bridge/Monolog/Processor/TokenProcessor.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Monolog\Processor;
+
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Role\Role;
+
+/**
+ * Adds the current security token to the log entry.
+ *
+ * @author Dany Maillard <danymaillard93b@gmail.com>
+ */
+class TokenProcessor
+{
+    private $tokenStorage;
+
+    public function __construct(TokenStorageInterface $tokenStorage)
+    {
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    public function __invoke(array $records)
+    {
+        if ($this->isDeniedProcess()) {
+            return $records;
+        }
+
+        $records['extra']['token'] = null;
+        if (null !== $token = $this->tokenStorage->getToken()) {
+            $records['extra']['token'] = array(
+                'username' => $token->getUsername(),
+                'authenticated' => $token->isAuthenticated(),
+                'roles' => array_map(function (Role $role) { return $role->getRole(); }, $token->getRoles()),
+            );
+        }
+
+        return $records;
+    }
+
+    public function isDeniedProcess()
```

Comment:
In CLI mode, ``extra.token`` should not be added, for example in commands but not in tests. This method allows [to mock this behavior](https://github.com/symfony/symfony/pull/21086/commits/653c24455d8689671e450f4c84b1d2bf017ab6b8#diff-e846e2de8865a11af799406eac1d2bdeR37) in tests.